### PR TITLE
JetBrains: Separate dotcom vs enterprise settings

### PR DESCRIPTION
**Pull request by vdavid**
Fixes https://github.com/sourcegraph/sourcegraph/issues/38575

This is what our Preferences look like now:

CleanShot 2022-07-15 at 15 36 34@2x

This change also fixes the URL-saving bug that Coury encountered yesterday. It was a silly one, basically a typo, and a one-line fix. (You can see it as a separate commit if interested.)

## Test plan

Made this [4-min Loom](https://www.loom.com/share/ddf27863a807413a83fdf204fc685c44) where I demo the UI and the functionality

## App preview:

- [Web](https://sg-web-dv-jetbrains-reorganize-settings.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-betsnabnvc.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by philipp-spiess** [comment]
The new settings screen looks really good now. Great work! 🚀

File: client/jetbrains/src/main/java/com/sourcegraph/config/NotificationActivity.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public void runActivity(@NotNull Project project) {
         if (lastNotifiedPluginVersion == null || lastNotifiedPluginVersion.compareTo(latestReleaseMilestoneVersion) < 0) {
             notifyAboutUpdate(project);
         } else {
-            String url = ConfigUtil.getSourcegraphUrl(project);
+            String url = ConfigUtil.getEnterpriseUrl(project);
             if (!ConfigUtil.isUrlNotificationDismissed() && (url.length() == 0 || url.startsWith("https://sourcegraph.com"))) {
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/config/NotificationActivity.java:35]: Totally agreed, and checked that it's safe to filter for the trailing slash here, so added that.

File: client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java [hunk 2]
```diff
@@ -50,17 +60,101 @@ public JPanel getPanel() {
...
+        // Assemble the two main panels
+        JBLabel dotComComment = new JBLabel("Use sourcegraph.com to search public code", UIUtil.ComponentStyle.SMALL, UIUtil.FontColor.BRIGHTER);
+        dotComComment.setBorder(JBUI.Borders.emptyLeft(20));
+        JPanel dotComPanel = FormBuilder.createFormBuilder()
+            .addComponent(sourcegraphDotComRadioButton, 1)
+            .addComponentToRightColumn(dotComComment, 2)
+            .getPanel();
+        JPanel enterprisePanelContent = FormBuilder.createFormBuilder()
+            .addLabeledComponent(urlLabel, urlTextField, 1)
+            .addTooltip("If your company has your own Sourcegraph instance, set its URL here")
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java:130]: Changed it to "its". I've also found it a bit weird, even though I wrote it. :)

File: client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java [hunk 3]
```diff
@@ -50,17 +60,101 @@ public JPanel getPanel() {
...
+            .addLabeledComponent(accessTokenLabel, accessTokenTextField, 1)
+            .addComponentToRightColumn(userDocsLinkComment, 1)
+            .addComponentToRightColumn(accessTokenLinkComment, 1)
+            .getPanel();
+        enterprisePanelContent.setBorder(IdeBorderFactory.createEmptyBorder(JBUI.insets(1, 30, 0, 0)));
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java:135]: Yeah, restarting the IDE each time was meh. I was very grateful for this beefy Mac.

File: client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java [hunk 4]
```diff
@@ -50,17 +60,101 @@ public JPanel getPanel() {
...
+        JPanel enterprisePanel = FormBuilder.createFormBuilder()
+            .addComponent(enterpriseInstanceRadioButton, 1)
+            .addComponent(enterprisePanelContent, 1)
+            .getPanel();
+
+        // Assemble the main panel
+        JPanel userAuthenticationPanel = FormBuilder.createFormBuilder()
+            .addComponent(dotComPanel)
+            .addComponent(enterprisePanel, 5)
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java:144]: Just another inset, like the others.

**Comment by vdavid** [comment]
Fixed everything, thanks for the review, @philipp-spiess! Then just waiting for a go-ahead or change requests from @jjinnii.

**Review by jjinnii — APPROVED** [review]
LGTM, a few minor copy callouts: 
Instead of "your own instance", it should be "If your company uses a private Sourcegraph instance"
Also regarding "you might need an access token to sign in", I believe we had said that we will go for the majority use case and change to "You will need an access token to sign in."

**Pull request merged**